Tests/NIOExtrasTests/LengthFieldBasedFrameDecoderTest.swift

```diff
@@ -455,4 +455,37 @@ class LengthFieldBasedFrameDecoderTest: XCTestCase {
             })
         }
     }
+    
+    func testMaliciousLengthOn32BitPlatform() throws {
+        self.decoderUnderTest = .init(LengthFieldBasedFrameDecoder(lengthFieldLength: .four,
+                                                                   lengthFieldEndianness: .little))
+        XCTAssertNoThrow(try self.channel.pipeline.addHandler(self.decoderUnderTest).wait())
+
+        let dataLength: UInt32 = UInt32(Int32.max) + 1
+        
+        var buffer = self.channel.allocator.buffer(capacity: 4) // 4 byte header
+        buffer.writeInteger(dataLength, endianness: .little, as: UInt32.self)
+        buffer.writeString(standardDataString)
+        if UInt32.bitWidth == UInt.bitWidth {
+            // LengthFieldBasedFrameDecoder.ByteLength.four should only throw if we are on a 32 bit platform
+            XCTAssertThrowsError(try self.channel.writeInbound(buffer))
+        } else {
+            XCTAssertFalse(try self.channel.writeInbound(buffer).isFull)
+        }
+    }
+    
+    func testMaliciousLengthOn64BitPlatform() {
+        // LengthFieldBasedFrameDecoder.ByteLength.eight is only supported on 64 bit systems
```

Comment:
oh, why is that? It should be supported everywhere. If not, that's a bug we should fix.